en/test_and_ci/continous_integration.md

```diff
@@ -1,3 +1,77 @@
+  - ... (other auxiliary tests)
+- `ekf_functional_change_indicator.yml`: Check if there's any functionality change in EKF ?
+- `ekf_update_change_indicator.yml`: Check if there's any functionality change in EKF ?
+- `failsafe_sim.yml`: Builds [failsafe simulator](../config/safety_simulation.md)
+- `clang-tidy.yml`: Runs Clang Tidy to check if coding style is consistent
+- `compile_linux.yml`: Builds linux flight controller board targets
+
+## PX4 release / main branch artifacts (+ Board specific files)
+Upon changes to the `main` branch, or to `release/*` branches (any releases), certain artifact files need to be built and uploaded to the server.
+This is handled by [this](https://github.com/PX4/PX4-Autopilot/blob/main/.github/workflows/deploy_all.yml) workflow.
+
+For each board targets, the artifacts built are:
+- Parameter metadata (`parameters.*`)
+- Events metadata (`events/*.xz`)
+- Actuators metadata (`actuators.json*`)
+- Target binary (`*.px4`)
+
+Then these files get uploaded to the [Amazon S3 bucket server](https://px4-travis.s3.amazonaws.com/).
+
+### Usage by QGC
+These files then get used in QGroundControl for:
+- Flashing PX4 binaries (stable: `release/*` / beta: `main`) through Firmware tab
+- Downloading & Caching the Events / Actuators / Parameters metadata for boards that doesn't support [MAVLink Component Information service](https://mavlink.io/en/services/component_information.html) based metadata retrieval
+
+Without these artifacts getting uploaded, QGroundControl would quickly go out of sync / won't be able to download & flash target binaries. So this is a crucial part of CI.
+
+## Generic metadata for Documentation & QGC by main branch (Not board specific)
+Whenever there are changes to the `main` branch, separately from the `deploy_all.yml` workflow above, Jenkins builds / deploys metadata.
+Jenkins file can be found [here](https://github.com/PX4/PX4-Autopilot/blob/main/Jenkinsfile).
+
+### PX4 Documentation update
+It uses the [PX4BuildBot](https://github.com/PX4BuildBot) account to upload the following metadata to update the documentation:
+- `airframes_reference.md`: Markdown for [Airframes Reference Documentation](../airframes/airframe_reference.md)
+- `parameter_reference.md`: Markdown for [Parameter Reference Documentation](../advanced_config/parameter_reference.md)
+- `modules/*.md`: Markdown for [Modules documentation](../modules/README.md)
+- `graph_*.json`: JSON defining uORB Graph, visualized in [uORB Graph](../middleware/uorb_graph.md) page
+- `msg_docs/*.md`: Markdown for [uORB Messages documentation](../msg_docs/README.md)
+- `failsafe_sim/*`: Files for [Failsafe Simulator Interface](../config/safety_simulation.md)
+
+An example commit can be found [here](https://github.com/PX4/PX4-user_guide/commit/a4ffaf5fc32df8650da614aa3129b3b90794584f).
+
+Without these metadata getting pushed, PX4 documentation will go out of sync very quickly! So it's a crucial piece of CI.
```

Comment:
FYI Everything is a crucial bit of CI. This kind of thing adds words to no value.